readme.md

```diff
@@ -723,10 +723,16 @@ its `url` and `title` defined somewhere else in the document by a
 `referenceType` is needed to detect if a reference was meant as a
 reference (`[foo][]`) or just unescaped brackets (`[foo]`).
 
+`reference` provides the original raw reference, if the `referenceType` is
+`full` and the reference differs from the `identifier`.  This enables
```

Comment:
Would it make sense to drop the case where `referenceType` is `full`? Although it’s not needed to create the original markdown for collapsed or shortcut references in remark, it’s one less condition in the spec?

Reply:
Maybe it could be useful for linting, or non-HTML and non-markdown outputs?

Reply:
* For the trade-off, I think that’s fine. How many references are out there really.
* For the test suite: the remarkjs repo has a script: `script/regenerate-fixtures` that updates the ASTs for all tests, and you can check the diff if it works or not. I personally don’t see the addition of a property to nodes as a breaking change. Even if someones tests break because of fixtures, I don’t think ~~actual~~ (not sure how to call it) code will break.
* And usefulness: I meant that the addition of a `reference` to non-full references could potentially be useful for other things, like linting, even though we’re now discussing stringifying to markdown and html, which doesn’t need it.